src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -1042,6 +1042,78 @@ namespace OpenLoco::Vehicles
         }
     }
 
+    // 0x004B1C48
+    // Applies the vehicle object lengths to the bogies of the train
+    // with some specified starting distance. Bodies are not set as
+    // they are calculated later based on the positions of the bogies.
+    // returns the final distance
+    static int32_t applyVehicleObjectLengthToBogies(Vehicle& train, const int32_t startDistance)
+    {
+        auto distance = startDistance;
+        for (auto& car : train.cars)
+        {
+            const auto* vehicleObj = ObjectManager::get<VehicleObject>(car.front->objectId);
+            assert(std::distance(car.begin(), car.end()) == vehicleObj->var_04);
+            if (car.body->has38Flags(Flags38::isReversed))
+            {
+                auto objCarIndex = vehicleObj->var_04 - 1;
+                for (auto& component : car)
+                {
+                    auto& objCar = vehicleObj->carComponents[objCarIndex];
+                    const auto frontLength = objCar.backBogiePosition * -2179;
```

Comment:
this 2179 is used a lot, can we put it in a constant?

Reply:
its the conversion rate to one of the higher precision distance types but at this time I'm not 100% sure.